iceoryx2-ffi/ffi/src/api/attribute.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#![allow(non_camel_case_types)]
+
+use iceoryx2::service::attribute::Attribute;
+
+use core::ffi::c_char;
+use std::ffi::CString;
+
+// BEGIN types definition
+pub struct iox2_attribute_h_t;
+
+impl iox2_attribute_h_t {
+    pub(crate) unsafe fn underlying_type(&self) -> &Attribute {
+        &*(self as *const iox2_attribute_h_t).cast()
+    }
+}
+
+pub type iox2_attribute_h_ref = *const iox2_attribute_h_t;
+
+// END type definition
+
+// BEGIN C API
+const ZERO_TERMINATOR_LEN: usize = 1;
+
+/// Returns the length of the attributes key.
+///
+/// # Safety
+///
+/// * The `handle` must be a valid handle.
+#[no_mangle]
+pub unsafe extern "C" fn iox2_attribute_key_len(handle: iox2_attribute_h_ref) -> usize {
+    debug_assert!(!handle.is_null());
+
+    let attribute = (*handle).underlying_type();
+    attribute.key().len() + ZERO_TERMINATOR_LEN
+}
```

Comment:
This is kind of counterintuitive on how, e.g. `strlen` works, which returns the length without the null terminator. Why did you choose to go this route?